File: lib/features/dashboard/domain/entities/storage_volume_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'storage_volume_info.freezed.dart';
part 'storage_volume_info.g.dart';

@freezed
class StorageVolumeInfo with _$StorageVolumeInfo {
  const factory StorageVolumeInfo({
    required String name,
    required String path,
    required int totalSpace,
    required int freeSpace,
    required bool isRemovable,
    required bool isPrimary,
  }) = _StorageVolumeInfo;

  factory StorageVolumeInfo.fromJson(Map<String, dynamic> json) =>
      _$StorageVolumeInfoFromJson(json);
}

File: android/app/src/main/kotlin/com/movixalabs/cleanmaster/models/StorageVolumeInfo.kt
package com.movixalabs.cleanmaster.models

data class StorageVolumeInfo(
    val name: String,
    val path: String,
    val totalSpace: Long,
    val freeSpace: Long,
    val isRemovable: Boolean,
    val isPrimary: Boolean
) {
    fun toMap(): Map<String, Any> = mapOf(
        "name" to name,
        "path" to path,
        "totalSpace" to totalSpace,
        "freeSpace" to freeSpace,
        "isRemovable" to isRemovable,
        "isPrimary" to isPrimary
    )
}
